### PR TITLE
chore: provide OSSF security-insight

### PR DESCRIPTION
**Pull request by mmorel-35**
Clomonitor requires a SECURITY-INSIGHTS.yml,
See https://clomonitor.io/projects/cncf/etcd#etcd_security

[CLOMonitor](https://clomonitor.io/projects/cncf/etcd)

This was done with the help of https://github.com/ossf/security-insights-spec/tree/main/specification

### CODE REVIEW

**Comment by jmhbnz** [comment]
> cc. @jmhbnz, thoughts on this?

Thanks @mmorel-35 for raising the proposal. I like the idea / intent. However I am not a fan of the implementation as it feels like more clutter in the project root directory. It's not clear to me the benefits, is there a specific problem we are trying to solve?  I would like to see atttention/tidy-up/improvement to what we already hold under `security/` in the first instance.

**Comment by ahrtr** [comment]
Thanks for your contribution, but please don't assume that others have the same background or did the same investigation/study as you. Please try to provide more context when you raise a PR.

It took me roughly 15 ~ 20 minutes to investigate&google the background, the following is what I got, for others reference,
- `clomonitor` is a project under cncf: https://github.com/cncf/clomonitor .   It is a tool that periodically checks open source projects repositories to verify they meet certain project health best practices.
- `clomonitor` refers to the [ossf](https://github.com/ossf) in terms of security.
- Based on the https://clomonitor.io/docs/topics/checks/#dependencies-policy, the `SECURITY-INSIGHTS.yml` should be located at the root of the repository. 
-  Please also refer to  https://github.com/ossf/security-insights-spec/blob/v1.0.0/specification.md
- https://www.trailofbits.com/ is also for fixing security problems. Please see my question in https://github.com/etcd-io/etcd/pull/18940#discussion_r1858171608

File: SECURITY-INSIGHTS.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    - https://github.com/etcd-io/etcd/blob/main/go.mod
+  sbom:
+    - sbom-url: https://github.com/etcd-io/etcd/blob/main/bill-of-materials.json
+distribution-points:
+  - https://github.com/etcd-io/etcd/releases
+documentation:
+  - https://etcd.io/docs/
+security-assessments:
+  - auditor-name: Trail of Bits
+    auditor-url: https://www.trailofbits.com/
```
[the remark this answers is not in the document]
Comment (jmhbnz) [reply SECURITY-INSIGHTS.yml:33]: I believe it's referring to the old 2020 audit? https://github.com/etcd-io/etcd/blob/main/security/SECURITY_AUDIT.pdf

[the remark this answers is not in the document]
Comment (ahrtr) [reply SECURITY-INSIGHTS.yml:33]: OK, thx. I missed the following field 

> auditor-report: https://github.com/etcd-io/etcd/blob/main/security/SECURITY_AUDIT.pdf

File: README.md [hunk 2]
```diff
@@ -8,7 +8,9 @@
 [![Godoc](http://img.shields.io/badge/go-documentation-blue.svg?style=flat-square)](https://godoc.org/github.com/etcd-io/etcd)
 [![Releases](https://img.shields.io/github/release/etcd-io/etcd/all.svg?style=flat-square)](https://github.com/etcd-io/etcd/releases)
 [![LICENSE](https://img.shields.io/github/license/etcd-io/etcd.svg?style=flat-square)](https://github.com/etcd-io/etcd/blob/main/LICENSE)
+[![OpenSSF Best Practices](https://www.bestpractices.dev/projects/3192/badge)](https://www.bestpractices.dev/projects/3192)
```
Comment (ahrtr) [inline README.md:11]: It's OK to keep this badge.

**Comment by ahrtr** [comment]
> It's not clear to me the benefit or improvement to our project security posture as a result of merging this pr.

We already have [OpenSSF Scorecard report](https://scorecard.dev/viewer/?uri=github.com/etcd-io/etcd), so we may not want to do this using another tool (clomonitor) again.

But it's OK to keep the OpenSSF Best Practices badge as mentioned in https://github.com/etcd-io/etcd/pull/18940#discussion_r1869069139

**Comment by ahrtr** [comment]
As mentioned above in https://github.com/etcd-io/etcd/pull/18940#issuecomment-2516725394, `OpenSSF Scorecard` and `CLOMonitor` are doing similar thing, so we might not want to integrate `CLOMonitor` for now.

**Pull request closed without merging**